plutus-core/plutus-ir/test/PlutusIR/Transform/LetFloatIn/Tests.hs

```diff
@@ -61,3 +65,13 @@ test_letFloatInRelaxed = runTestNestedIn ["plutus-ir/test/PlutusIR/Transform/Let
         _ <- runQuoteT . flip inferType (() <$ pirFloated) =<< TC.getDefTypeCheckConfig ()
         -- letmerge is not necessary for floating, but is a nice visual transformation
         pure $ LetMerge.letMerge pirFloated
+
+-- | Check that a term typechecks after a
+-- `PlutusIR.Transform.LetFloatIn.floatTerm` pass.
+typecheck_floatTerm_prop :: Property
+typecheck_floatTerm_prop =
+  non_pure_typecheck_prop $ LetFloatIn.floatTerm def True
```

Comment:
I think we maybe _do_ want to test both ways with the conservative flags for various passes, though! In particular, we can easily generate a random bool and use that as the argument.

Reply:
OK I tried just letting `tasty-discover` detect this prop with `Bool` as argument. The default number of tests is only 100, which is way too small. I just wrote it out again - that way we can see which mode fails which is useful.

Reply:
Then I do think this is a case to run with more tests. We should just need twice as many tests!